[common] Add basic bit operations

VasiliyMatr opened the pull request:
Add basic bit operations into common module

VasiliyMatr committed:
[common] Add basic bit operations

TheRedHotHabanero commented on common/include/sim/common.hpp:
```diff
@@ -15,6 +29,67 @@ using VirtAddr = RegValue;
+static_assert(CHAR_BIT == BYTE_SIZE);
+
+// Return ones mask for given unsigned integer type
+template <class UInt> constexpr UInt ones() noexcept {
```
please change func name
For exmaple, onesMask

VasiliyMatr replied:
Done

VasiliyMatr committed:
[cmake][common] Add tests for common::bit funcs

VasiliyMatr committed:
[common] Fix bugs in common::bit

- Fix signExtend and getBits functions

VasiliyMatr committed:
[common] Rename bit::ones -> bit::onesMask

- New name is clearer

VasiliyMatr committed:
[common] Refactor bit operations

- Rename: getBitsShifted -> getBitField
- Rename: getBits -> maskBits
- Refactor getBitField and maskBits bodies
- Refactor signExtend bodies
- Update bit tests

TheRedHotHabanero commented on common/tests/src/test_bit.cpp:
```diff
@@ -0,0 +1,49 @@
+namespace sim {
+namespace bit {
+
+TEST(Bit, ones) {
```
minor: ones -> onesMask
just to use common style and naming

VasiliyMatr committed:
[common] Fix bit::onesMask test name
